### ISSUE TITLE
smv-shell overwrites Python reserved globals that collide with pyspark function names

### ISSUE THREAD

**Issue by jacobdr** [TresAmigosSD/SMV#950]
The line [here](https://github.com/TresAmigosSD/SMV/blob/master/tools/conf/smv_pyshell_init.py.template#L51) does a wildcard import of pyspark.sql.functions. When there is a collision between the name of a pyspark function and a reserved global python keyword (such as [built-in functions](https://docs.python.org/2/library/functions.html)) -- for example, max and min -- the pyspark function wins out. This leads to some strange error messages in the smv-pyshell and therefore Jupyter notebooks:

```python
list_of_numbers = list(range(10))
max(list_of_numbers)
```
Expected output: `9`

Observed behavior:
```python
Py4JError Traceback (most recent call last)
<ipython-input-255-60abc40299af> in <module>()
      1 list_of_numbers = list(range(10))
----> 2 max(list_of_numbers)

/usr/hdp/2.5.0.0-1245/spark/python/pyspark/sql/functions.py in _(col)
     37     def _(col):
     38         sc = SparkContext._active_spark_context
---> 39         jc = getattr(sc._jvm.functions, name)(col._jc if isinstance(col, Column) else col)
     40         return Column(jc)
     41     _.__name__ = name

/usr/hdp/2.5.0.0-1245/spark/python/lib/py4j-0.9-src.zip/py4j/java_gateway.py in __call__(self, *args)
    811         answer = self.gateway_client.send_command(command)
    812         return_value = get_return_value(
--> 813             answer, self.gateway_client, self.target_id, self.name)
    814 
    815         for temp_arg in temp_args:

/usr/hdp/2.5.0.0-1245/spark/python/pyspark/sql/utils.py in deco(*a, **kw)
     43     def deco(*a, **kw):
     44         try:
---> 45             return f(*a, **kw)
     46         except py4j.protocol.Py4JJavaError as e:
     47             s = e.java_exception.toString()

/usr/hdp/2.5.0.0-1245/spark/python/lib/py4j-0.9-src.zip/py4j/protocol.py in get_return_value(answer, gateway_client, target_id, name)
    310                 raise Py4JError(
    311                     "An error occurred while calling {0}{1}{2}. Trace:\n{3}\n".
--> 312                     format(target_id, ".", name, value))
    313         else:
    314             raise Py4JError(

Py4JError: An error occurred while calling z:org.apache.spark.sql.functions.max. Trace:
py4j.Py4JException: Method max([class java.util.ArrayList]) does not exist
	at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:335)
	at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:360)
	at py4j.Gateway.invoke(Gateway.java:254)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:209)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by jacobdr** [comment]
@AliTajeldin @ninjapapa @laneb This might be a painful one to fix... The fact that it only affects the shell means that no production code should get broken, but hard to measure. Because we just announced the semver breaking change with the SMV 2.x line anyway, we should probably try to patch this one up sooner rather than later.

**Comment by ninjapapa** [comment]
It is an issue we need to be consistent about. Current use case have pyshell as ad hoc and code as persisted for reuse. Therefore to make the Sql functions easy to use in shell make sense. 

But in the future, when code in shell and code in file are more interchangeable, current approach is problematic.

Agree to change it. A way to allow users who still want to use those functions directly is to print a note when start shell to point out user can add their own init script and import.

**Comment by jacobdr** [comment]
I think our examples should switch to the "functions as F, types as T" import style. Wildcard * imports are an anti pattern, and most Python linters and type checkers will complain. Two character difference shouldn't create much hassle on the user.

**Comment by laneb** [comment]
Agreed on removing wildcard imports. But why not omit these two imports altogether and just make users explicitly import the functions they want? The user can make any aliases they want in their init script.

**Comment by AliTajeldin** [comment]
For shell, I like the "as F" style.  It is a pain for the user to re-import the sql functions every time they need to use them in shell.  For code, we should make the user do the import explicitly.

**Comment by laneb** [comment]
Fixed by #965.
